### PR TITLE
Change text edit implementation to support \r

### PR DESCRIPTION
**Pull request by hrsh7th**
This PR aims to support `\r\n`, `\r` and `\n` in text edits.

The approach is using only `\n` and modification uses `setline`.

### CODE REVIEW

**Comment by hrsh7th** [comment]
This implementation was ported by `hrsh7th/vim-vital-vs`.
It passed vscode's text edit test cases.

This PR should fix https://github.com/prabirshrestha/vim-lsp/issues/659.

**Comment by hrsh7th** [comment]
I think it is ok to merge.

File: autoload/lsp/utils.vim [hunk 1]
```diff
@@ -319,3 +319,11 @@ function! lsp#utils#make_valid_word(str) abort
    endif
    return l:str
 endfunction
+
+function! lsp#utils#_split_by_eol(text) abort
+    let l:text = a:text
+    let l:text = substitute(l:text, "\r\n", "\n", 'g')
+    let l:text = substitute(l:text, "\r", "\n", 'g')
+    return split(l:text, "\n", v:true)
```
Comment (mattn) [inline autoload/lsp/utils.vim]: Equivarent to this?

```
return split(a.text, '\r\n\|\r', 1)
```

Comment (hrsh7th) [reply autoload/lsp/utils.vim]: Thanks for your review.
test case can pass with `split(a:text, '\r\n\|\r\|\n', v:true)`.

I will change to it.

**Comment by hrsh7th** [comment]
This approach is explained following.

```
abcdefg
hijklmn
opqrstu
```

If text edit is below

- range: `efg -> opq`
- newText: "000\n111\n222"

new implementation will create `l:new_lines` like below

```
abc000
111
222rstu
```

and then append `l:new_lines` simply.

**Comment by mattn** [comment]
Thank you.

**Pull request merged**